viz_scripts/survey_responses.ipynb

```diff
@@ -322,45 +326,56 @@
    },
    "outputs": [],
    "source": [
-    "#create one plot per question in the survey\n",
+    "#create one plot per survey, one bar per question\n",
     "for survey_name in quest_dict.keys():\n",
     "    print(\"Charts for:\", survey_name)\n",
     "#     debug_df = generate_debug_df(program, include_test_users, total_dfs[survey_name], total_dfs[survey_name][total_dfs[survey_name]['user_input'] != {} if len(total_dfs[survey_name]) > 0 else pd.DataFrame()])\n",
     "    debug_df = generate_debug_df(program, include_test_users, all_confirmed_trips, df_responses) #survey filtering still buggy, omitting all features of \"num trips survey presented\"\n",
     "    \n",
-    "    for col in quest_dict[survey_name].keys():\n",
-    "        filename = col + file_suffix\n",
-    "        plot_title_no_quality = survey_name + \"\\n\" + quest_dict[survey_name][col]\n",
+    "    n =len(quest_dict[survey_name].keys())\n",
+    "    fig, ax = plt.subplots(nrows=n, ncols=1, figsize=(15,2*n), sharex=True)\n",
+    "    text_results = [[\"Unmodified Alt Text\", \"Unmodified HTML\"]] * n\n",
+    "    filename = survey_name + file_suffix\n",
+    "    \n",
+    "    #temp quality text -- just based on the responses\n",
+    "    plot_title_no_quality = survey_name\n",
+    "    \n",
+    "    try:\n",
+    "        qual_text = scaffolding.get_quality_text_numerator(df_responses[df_responses['survey_name'] == survey_name], include_test_users) if 'survey_name' in df_responses.columns else \"No Responses\"\n",
+    "        plot_title = plot_title_no_quality+'\\n'+qual_text\n",
+    "    \n",
+    "        for i in range(len(quest_dict[survey_name].keys())):\n",
+    "            col = list(quest_dict[survey_name].keys())[i]\n",
```

Comment:
not quite sure what this does - are we doing

it looks like we are iterating through the survey names, but then isn't `col` just the question name? A more descriptive variable name would be helpful! Also I think that this can be simplified as

```suggestion
    "        for question_name in quest_dict[survey_name].keys():\n",
```

Reply:
I use the `i` to control which position each bar takes - to add all the questions from a given survey to the same chart

I've updated the variable `n` to a more descriptive name `n_questions` and reused it to set the number of inner loops needed

I updated `col` to `quest_name` to be more descriptive